src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php

```diff
@@ -198,7 +198,7 @@ public function resolveValue(mixed $value, array $resolving = []): mixed
             return $args;
         }
 
-        if (!\is_string($value) || 2 > \strlen($value)) {
+        if (!\is_string($value) || 2 > \strlen($value) || !str_contains($value, '%')) {
```

Comment:
let's optimize for longer strings, which should be the most common:
```suggestion
        if (!\is_string($value) || '' === $value || !str_contains($value, '%')) {
```